vnext/Microsoft.ReactNative.Managed.CodeGen/Model/ReactMethod.cs

```diff
@@ -13,21 +15,31 @@ public class ReactMethod
 
     public bool IsSynchronous { get; }
 
-    public MethodReturnType ReturnType { get; }
+    public MethodReturnStyle ReturnStyle { get; }
 
-    public ReactMethod(IMethodSymbol method, string name, bool isSynchronous = false)
+    public ITypeSymbol EffectiveReturnType { get; }
+
+    public IReadOnlyList<IParameterSymbol> EffectiveParameters { get; }
+
+    public ReactMethod(IMethodSymbol method, string name, MethodReturnStyle returnStyle, ITypeSymbol effectiveReturnType, IReadOnlyList<IParameterSymbol> effectiveParameters, bool isSynchronous)
     {
+      Contract.Requires(! (IsSynchronous && returnStyle == MethodReturnStyle.Task), "Task style methods are required to be asynchronous");
       Method = method;
       Name = name;
+      ReturnStyle = returnStyle;
+      EffectiveReturnType = effectiveReturnType;
+      EffectiveParameters = effectiveParameters;
       IsSynchronous = isSynchronous;
     }
 
-    public enum MethodReturnType
+    public enum MethodReturnStyle
     {
       Void = 0,
```

Comment:
> = 0 [](start = 10, length = 4)

I wonder why is needed to specify values explicitly?

Reply:
Just a habit from me working with flags, and compatibility since at runtime they are just integer values, this makes it clear adding new values in the middle is not good because you have to renumber and are breaking binary version compatilbity...